### DISCUSSION TITLE
[Chapter 03] final 필드의 단점은 없나?

### DISCUSSION THREAD

**Discussion by 0sunset0** [cowaine/is-my-code-that-strange#14] [Chapter 01-03 (설계)] [answered]
```java
Money add(final int other) {...}
```
코드 3.12에서는 메서드 매개변수를 `final`로 선언하고 있습니다. 

근데 `final`로 선언하는 것이 귀찮기도 하고 (인텔리제이의 도움을 받으면 되지만)
여러 사람이 같이 코드를 작업할 때 어떤 매개변수는 `final` 키워드가 누락되어 통일성이 없어질 수도 있고, 매개변수 선언 코드가 길어져
`굳이 이렇게 해야 하나?`라는 생각이 들었습니다.

단점보다 장점이 더 많을 때 사용하면 될텐데, 그럼 final의 단점에는 무엇이 있고 final을 안쓰는 게 더 좋은 상황에는 구체적으로 무엇이 있을까요?

**Comment by firewoody237** [comment, marked as answer]
이거는 찬성파/반대파가 있더라구요.
- 찬성파 : 안정성을 위해 하는 것이 좋다.
- 반대파 : 코드 길어져서 가독성 떨어지는데 굳이 이렇게 해야돼?

이성적으로는 두 내용을 비교해 봤을 때 쓰는게 좀 더 좋지 않나 싶습니다.

저는 반대에 가까운 중립파인데, 팀 컨벤션이면 쓰지만 그 외에는 잘 안쓰고 있습니다.
뭔가.. 메서드 시그니처가 길어지는 착각이 들어 코드가 안예쁘더라구요 ㅋㅋ

**Reply by firewoody237** [reply]
위는 제 의견이 였고..
4장을 추가적으로 읽어본 결과
final을 안쓰는게 좋은 상황은 "성능을 신경서야 할 때"와 "스코프가 국소지역일 때" 라고 합니다.

다양한 언어들이 immutable한 상태를 디폴트로 내놓는다고 하는 것을 보면.. 특별한 이유가 있지 않은 이상 쓰는게 좋아보이네요 ㅎㅎ

**Comment by Irisation23** [comment]
https://stackoverflow.com/questions/500508/why-should-i-use-the-keyword-final-on-a-method-parameter-in-java

**Comment by corock** [comment]
예고르 부가옌코([_Yegor Bugayenko_](https://github.com/yegor256))라는 분이 좋은 객체의 7가지 덕목([_Seven Virtues of a Good Object_](https://www.yegor256.com/2014/11/20/seven-virtues-of-good-object.html)) 이라는 글을 포스팅한 적이 있습니다.
여기서 좋은 객체는 '객체의 클래스가 `final` 이나 `abstract` 이다' 라고 제시하는 부분이 있는데, 이 부분을 함께 참고하면 좋을 것 같습니다.
